src/libs/actions/Policy.ts

```diff
@@ -2590,6 +2600,336 @@ function clearCategoryErrors(policyID: string, categoryName: string) {
     });
 }
 
+function enablePolicyCategories(policyID: string, isEnabled: boolean) {
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    areCategoriesEnabled: isEnabled,
+                    pendingFields: {
+                        areCategoriesEnabled: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    pendingFields: {
+                        areCategoriesEnabled: null,
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    areCategoriesEnabled: !isEnabled,
+                    pendingFields: {
+                        areCategoriesEnabled: null,
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters: EnablePolicyCategoriesParams = {policyID, isEnabled};
+
+    API.write(WRITE_COMMANDS.ENABLE_POLICY_CATEGORIES, parameters, onyxData);
+
+    if (isEnabled) {
+        const navigationAction = getIsNarrowLayout() ? Navigation.goBack : () => Navigation.navigate(ROUTES.WORKSPACE_CATEGORIES.getRoute(policyID));
```

Comment:
I think the `backTo` is not the best solution here. If user deeplinks to the more features page on mobile from a helpsite for example then enabling a feature will take them to wrong page. 

I think we need to use the Navigation.navigate with FORCED_UP style to replace the page with workspace settings page https://github.com/Expensify/App/blob/2c5d6c54d4c216746325754211af43ceeae14102/src/libs/actions/Report.ts#L1128

Reply:
Oh that's a good catch!